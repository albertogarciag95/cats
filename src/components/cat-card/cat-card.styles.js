import styled from 'styled-components';

export const Card = styled.div`
  flex: 0 1 40%;
  margin: 20px;
  padding: 10px;
  text-align: left;
  border-radius: 5px;
  border: 1px solid black;
  background-color: white;
  display: inline-flex;
  flex-direction: column;
  justify-content: space-between;
  min-width: 400px;
`;

export const CardContent = styled.div`
  display: inline-flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const CardImage = styled.div`
  background-image: url(${props => props.image});
  background-size: contain;
  width: 100%;
  height: 100px;
`;

export const CardDescription = styled.span`
  margin-top: 10px;
  height: 100%;
`;

export const CardLink = styled.a`
  margin-top: 10px;
  color: blue;
  cursor: pointer;
`;