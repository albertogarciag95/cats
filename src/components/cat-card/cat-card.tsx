import React, { useEffect, useState } from 'react';
import { 
  Card, 
  CardContent, 
  CardDescription, 
  CardLink, 
  CardImage, 
  CardImageWrapper, 
  MAX_DESCRIPTION_LENGTH 
} from './cat-card.styles';

type CatCardProps = {
  description: string
};

export const CatCard = ({ description }: CatCardProps) => {
  const [imageUrl, setImageUrl] = useState('')

  useEffect(() => {
    fetch('https://cataas.com/cat?width=500', { cache: 'reload' })
      .then((image: Response) => image.blob())
      .then((blob: Blob) => URL.createObjectURL(blob))
      .then((url: string) => setImageUrl(url))
  }, [description])

  return (
    <Card>
        <CardImageWrapper>
          <CardImage alt='cat' src={imageUrl} />
        </CardImageWrapper>
        <CardContent>
          <CardDescription>{description}</CardDescription>
          <CardLink 
            isShown={description.length > MAX_DESCRIPTION_LENGTH}>
              Read more
          </CardLink>
        </CardContent>
    </Card>
  );
};