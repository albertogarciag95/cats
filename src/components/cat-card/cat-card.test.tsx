import renderer from 'react-test-renderer';
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from 'enzyme';

import { CatCard } from './cat-card';

configure({ adapter: new Adapter() });

const props = {
  description: 'test'
}

test('CatCard match snapshot', () => {
  const component = renderer.create(
    <CatCard {...props}/>
  );
  let cardlink = component.toJSON();
  expect(cardlink).toMatchSnapshot();
});

test('CardContent props verified', () => {
  const component = shallow(<CatCard {...props}/>)
  expect(component.find('CardImage').props().src).toBe('');
  expect(component.find('CardDescription').props().children).toBe('test');
  expect(component.find('CardLink').props().description).toBe('test');
});
