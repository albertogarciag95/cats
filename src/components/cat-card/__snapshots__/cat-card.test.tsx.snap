// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatCard match snapshot 1`] = `
<div
  className="sc-bdnxRM jOCwi"
>
  <div
    className="sc-hKFxyN gcWuxB"
  >
    <img
      alt="cat"
      className="sc-dlnjwi ibQVqM"
      src=""
    />
  </div>
  <div
    className="sc-gtsrHT jDeUIw"
  >
    <span
      className="sc-eCApnc drERXz"
    >
      test
    </span>
    <a
      className="sc-jSFjdj jAkyBl"
      onMouseEnter={[Function]}
    >
      Read more
    </a>
  </div>
</div>
`;
